### PR TITLE
Default timeout for Research is None; RC can be without name

### PR DESCRIPTION
**Pull request by AlexeyKozhevin**
The previous default value for timeout (5 minutes) in Research doesn't allow to execute massive tasks in callables.
Now by default timeout is None.

RC (ResearchConfig) named expression can be used without the specification of the executable unit.

### CODE REVIEW

File: batchflow/research/workers.py [hunk 1]
```diff
@@ -107,14 +107,15 @@ def __call__(self, queue, results):
...
-                            if signal is None and (time.time() - last_update_time.value) / 60 > self.timeout:
-                                p = psutil.Process(pid)
-                                p.terminate()
-                                message = 'Job {} [{}] failed in {}'.format(job[0], pid, self.worker_name)
-                                self.logger.info(message)
-                                final_signal.exception = TimeoutError(message)
-                                results.put(copy(final_signal))
-                                break
+                            if signal is None:
+                                if self.timeout is not None and (time.time() - last_update_time.value) / 60 > self.timeout:
```
Comment (a-arefina) [inline batchflow/research/workers.py:111]: it is better to unite these 2 conditions into one

File: batchflow/research/named_expr.py [hunk 2]
```diff
@@ -24,12 +24,11 @@ def get(self, **kwargs):
             _experiment = experiment
         else:
             _experiment = [experiment]
-        if self.name is not None:
-            res = [item[self.name] for item in _experiment]
-            if len(_experiment) == 1:
-                return res[0]
-            return res
-        return experiment
+        name = self.name if self.name is not None else list(_experiment[0].keys())[0]
```
Comment (a-arefina) [inline batchflow/research/named_expr.py:27]: This way of getting a missing name seems rather obscure. Is the first key in the first experiment that is passed to eval_expr is always the name we want? Or they are all equal?

Comment (AlexeyKozhevin) [reply batchflow/research/named_expr.py:27]: Actually, all configs are the same for all units

File: batchflow/research/workers.py (hunk 1 above)
Comment (AlexeyKozhevin) [reply batchflow/research/workers.py:111]: I would like to keep them separate because there are condition branches that only deal with signal conditions.

**Review by a-arefina — COMMENTED** [review]
It would be great if you write some tests, at least for `RC`

File: batchflow/research/workers.py [hunk 3]
```diff
@@ -107,14 +107,15 @@ def __call__(self, queue, results):
...
+                                    p = psutil.Process(pid)
+                                    p.terminate()
+                                    message = 'Job {} [{}] failed in {}'.format(job[0], pid, self.worker_name)
```
Comment (SergeyTsimfer) [inline batchflow/research/workers.py:114]: Please, change the message to something more descriptive like `failed because of timeout`

**Review by a-arefina — APPROVED** [review]

**Pull request merged**